web/docs/setup-guides/size-a-node.md

```diff
@@ -4,8 +4,69 @@ sidebar_position: 0
 
 # Size a node
 
-To better understand what resources you need to run a node, we provide the
-following calculator to derive **CPU**, **RAM**, and **storage** requirements.
+To better understand what resources you need to run a node, we provide guidance
+on sizing and a [calculator](#calculator) to derive concrete **CPU**, **RAM**,
+and **storage** requirements.
+
+## Considerations
+
+Several factors have an impact on sizing. Since you can run many types of
+workloads in pipelines, it is difficult to make a one-size-fits-all
+recommendation. The following considerations affect your resource requirements:
+
+### Workloads
+
+Depending on what you do with pipelines, you may generate a different resource
+profile.
+
+#### Filtering
+
+A typical [shaping](../user-guides/shape-data/README.md) operation is applying
+filters on the input values, e.g., by specifying a regular expression to only
+keep matching events. Filtering is a **CPU**-intensive operation. Unless you use
+numerous complex regular expressions, filtering is a cheap.
+  
+#### Aggregation
+
+Performing in-stream or historical aggregations often requires extensive
+buffering of the aggregation groups, which adds to your **RAM** requirements. If
+you run intricate custom aggregation functions, you also may see an additional
+increase CPU usage.
+
+#### Enrichment
+
+[Enriching](../user-guides/enrich-with-threat-intel/README.md) dataflows with
+[contexts](../contexts.md) requires holding in-memory state proportional to the
+context size. Therefore, enrichment affects your **RAM** requirements. [Bloom
+filters](../contexts/bloom-filter.md) are a fixed-size space-efficient structure
+for representing large sets, and [lookup tables](../contexts/lookup-table.md)
+grow linearly with the number of entries.
+
+### Data Diversity
+
+The more data sources you have, the more pipelines you run. In the simplest
+scenario where you just [import all data into a
+node](../user-guides/import-into-a-node/README.md), you deploy one pipeline per
+data source. The number of data sources is a thus a lower bound for the number
+of pipelines.
+
+### Data Volume
+
+The throughput of pipeline has an impact on performance. Pipelines with low data
+volume do not strain the system much, but high-volume pipelines substantially
+affect CPU and RAM requirements. Therefore, understanding your ingress volume,
+either as events per second or bytes per day, is helpful for sizing your node
+proportionally.
+
+### Retention
+
+When you leverage the node's built-in storage engine by
+[importing](../user-guides/import-into-a-node/README.md) and
+[exporting](../user-guides/export-from-a-node/README.md) data, you need
+persistent **storage**. To assess your retention span, you need to understand
+your data volume and your capacity.
```

Comment:
We should also mention that memory is needed for optimal query and automated lookup performance.

Reply:
Would "historical queries" be a separate workload then?

Reply:
Or maybe "import/export"

Reply:
Something like
```
### Lookups and Interactive Queries

You can improve the performance of historical query workloads with in-memory indexes.
In case you need to perform historical investigations regularly or want to use automatic lookups for IoCs you may want to provide additional memory to make use of these indexes.

Reply:
So far all the headings in this document are a *one-to-one* mapping to the headings in the calculator. I'm hesitant to add a new heading that is not represented in the calculator.

Reply:
Makes sense, will add that.